### PR TITLE
Adds RID arm64 to Testing runtime.json

### PR DESCRIPTION
**Pull request by ianhays**
unblocks building CoreFX tests for ARM64.

### CODE REVIEW

**Comment by ianhays** [comment]
@mellinoe @karajas

File: src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets [hunk 1]
```diff
@@ -15,7 +15,8 @@
   <PropertyGroup>
     <PrepareForRunDependsOn Condition="'$(IsTestProject)'=='true'">$(PrepareForRunDependsOn);CopyTestToTestDirectory;CreateTestNuGetPackage;ArchiveTestBuild</PrepareForRunDependsOn>
     <TestArchitecture Condition="'$(TestArchitecture)' == ''">x64</TestArchitecture>
-    <TestNugetRuntimeId Condition="'$(TestNugetRuntimeId)' == ''">win7-$(TestArchitecture)</TestNugetRuntimeId>
+    <TestNugetRuntimeId Condition="'$(TestNugetRuntimeId)' == '' and '$(TestArchitecture)'!='arm64'">win7-$(TestArchitecture)</TestNugetRuntimeId>
+    <TestNugetRuntimeId Condition="'$(TestNugetRuntimeId)' == '' and '$(TestArchitecture)'=='arm64'">win10-$(TestArchitecture)</TestNugetRuntimeId>
```
Comment (karajas) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets]: You could always pass in /p:TestNugetRuntimeId=win10-arm64.

Comment (ianhays) [inline src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets]: I actually ended up doing that anyways. You're right, this isn't necessary. I'll revert that bit.

File: src/Microsoft.DotNet.BuildTools.TestSuite/runtime.json [hunk 2]
```diff
@@ -69,6 +69,7 @@
             "netcoreapp1.0": [
                 "win7-x86",
                 "win7-x64",
+                "win10-arm64",
```
Comment (karajas) [inline src/Microsoft.DotNet.BuildTools.TestSuite/runtime.json]: Just wanted to confirm that this is the root of the arm64 leg of windows from https://github.com/dotnet/corefx/blob/master/pkg/Microsoft.NETCore.Platforms/runtime.json#L68-L69 .

Comment (ianhays) [inline src/Microsoft.DotNet.BuildTools.TestSuite/runtime.json]: yup!

**Comment by ianhays** [comment]
@ericstj @weshaggard @stephentoub

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets [hunk 3]
```diff
@@ -841,7 +841,7 @@
   <ItemGroup>
     <!-- Default validation frameworks : every framework that supports contracts -->
     <DefaultValidateFramework Include="netcoreapp1.0">
-      <RuntimeIDs>win7-x86;win7-x64;osx.10.11-x64;centos.7-x64;debian.8-x64;linuxmint.17-x64;opensuse.13.2-x64;rhel.7.2-x64;ubuntu.14.04-x64;ubuntu.16.04-x64</RuntimeIDs>
+      <RuntimeIDs>win7-x86;win7-x64;win10-arm64;osx.10.11-x64;centos.7-x64;debian.8-x64;linuxmint.17-x64;opensuse.13.2-x64;rhel.7.2-x64;ubuntu.14.04-x64;ubuntu.16.04-x64</RuntimeIDs>
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets]: Do you really want this for netcoreapp1.0 and not netcoreapp1.1?

Comment (ianhays) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets]: >Do you really want this for netcoreapp1.0 and not netcoreapp1.1?

no, I do not. I rebased on the latest master and added `win10-arm64` to netcoreapp1.1 instead.

I'll hold off on merging this PR until the solution to https://github.com/dotnet/corefx/issues/10335 comes.

Comment (ianhays) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets]: @karajas and I chatted offline about this and decided I should add netcoreapp1.1 to netcoreapp1.0 for forwards compat to unblock CoreFX, and he'll replace it later when dotnet/corefx#10335 comes. That way we can avoid modifying every test project.json in CoreFX more than once.

**Comment by ericstj** [comment]
LGTM

**Pull request merged**